src/components/core/tooltip/arrow-tooltip/arrow-tooltip.tsx

```diff
@@ -0,0 +1,206 @@
+                left: 'calc(100% - 16px)',
+            },
+            right_top: {
+                transform: 'rotate(180deg)',
+                bottom: 'calc(100% - 16px)',
+                left: 'calc(100% + 8px)',
+            },
+            right_center: {
+                transform: 'rotate(90deg) translate(-50%, 0%)',
+                top: '50%',
+                left: 'calc(100% + 8px)',
+                translateY: '50%',
+            },
+            right_bottom: {
+                transform: 'scaleX(-1)',
+                top: 'calc(100% - 16px)',
+                left: 'calc(100% + 8px)',
+            },
+            bottom_right: {
+                transform: 'rotate(270deg)',
+                top: 'calc(100% + 8px)',
+                left: 'calc(100% - 16px)',
+            },
+            bottom_center: {
+                transform: 'rotate(180deg) translate(-50%, 0%)',
+                top: 'calc(100% + 8px)',
+                right: '50%',
+            },
+            bottom_left: {
+                transform: 'scaleX(-1) rotate(270deg)',
+                top: 'calc(100% + 8px)',
+                right: 'calc(100% - 16px)',
+            },
+            left_top: {
+                transform: 'scaleX(-1) rotate(180deg)',
+                bottom: 'calc(100% - 16px)',
+                right: 'calc(100% + 8px)',
+            },
+            left_center: {
+                transform: 'rotate(270deg) translate(-50%, 0%)',
+                bottom: '50%',
+                right: 'calc(100% + 8px)',
+                translateY: '50%',
+            },
+            left_bottom: {
+                top: 'calc(100% - 16px)',
+                right: 'calc(100% + 8px)',
+            },
+        },
+    },
+});
+
+const Flex = styled('div', {
+    display: 'flex',
+});
+
+const ArrowTooltip = ({
+    arrow_direction = 'top_center',
+    arrow_color = '$coral500',
+    class_name,
+    class_name_arrow,
+    icon,
+    is_fixed_width,
+    children,
+    open_tooltip = true,
+    arrow_size,
+}: ArrowTooltipProps) => {
+    const { isDark } = useTheme();
+    const getArrowIcon = () => {
+        if (['top_center', 'right_center', 'bottom_center', 'left_center'].includes(arrow_direction))
+            return { icon: StraightArrowIcon, viewBox: '0 0 26 40' };
+
+        return { icon: CurvedArrowIcon, viewBox: '0 0 40 40' };
+    };
+    return (
+        <>
+            {open_tooltip && (
+                <TooltipContainer
+                    dark={isDark}
+                    css={{
+                        width: is_fixed_width ? 280 : 'fit-content',
+                        '@mobile': {
+                            width: is_fixed_width ? 200 : 'fit-content',
+                        },
+                    }}
+                    className={classNames(class_name)}
+                >
+                    <Flex>
+                        {icon && <Icon src={icon} />}
+                        <div>{children}</div>
```

Comment:
do we need the `div` surrounding the `children`?

Reply:
yeah here we will need as both icon and children needs to be in a row

Reply:
thanks thanks, got it :pray: